File: Level 002/2.py
def fib(maxValue):
	values = [1,1]
	while True:
		total = values[-1] + values[-2]
		if total > maxValue:
			break
		values.append(total)
	return values


def level2():
	return sum([x for x in fib(4000000) if x % 2 == 0])


if __name__ == "__main__":
	print level2()

File: Level 002/2.hs
fib :: Int -> Int
fib 0 = 0
fib 1 = 1
fib n = fib (n-1) + fib (n-2)

fibList :: [Int]
fibList = map fib [1..] --Woo! Super inefficiency!

result = sum $ filter even $ takeWhile (<4000000) fibList
